include/splitvector/split_tools.h

```diff
@@ -613,204 +600,300 @@ void split_prefix_scan_raw(T* input, T* output, splitStackArena& mPool, const si
    }
 }
 
-
-
-#ifdef __NVCC__
 template <typename T, typename Rule> 
 __global__ void block_compact(T* input,T* output,size_t inputSize,Rule rule,uint32_t *retval) 
 {
+   // This must be equal to at least both WARPLENGTH and MAX_BLOCKSIZE/WARPLENGTH
    __shared__ uint32_t warpSums[WARPLENGTH];
    const size_t tid = threadIdx.x + blockIdx.x * blockDim.x;
-   const size_t wid = tid /WARPLENGTH;
+   const size_t wid = tid / WARPLENGTH;
    const size_t w_tid = tid % WARPLENGTH;
+   const uint warpsPerBlock = Hashinator::defaults::MAX_BLOCKSIZE / WARPLENGTH;
```

Comment:
Hashinator::defaults is not accessible in this context. Mainly because we want to be able to use splitvectors separately. Tests do not compile.

Reply:
So the max blocksize can go in as a default template argument as is done in other split methods